File: game/forest/hurt_rock_hit_animation.py
from animation import Animation
from forest.forest_resources import ForestResources
from forest.hurt_rock_talking import HurtRockTalking
from render_type import RenderType
from state import State


class HurtRockHitAnimation(State):
    def __init__(self, context):
        super().__init__(context)
        self.needs_bottom = RenderType.POST

    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.hit_rock):
            return HurtRockTalking

        return None

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.hit_rock, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtRockHitAnimation.ts
/** Port of game/forest/hurt_rock_hit_animation.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { ForestResources } from './forestResources';
import { HurtRockTalking } from './hurtRockTalking';
import { RenderType } from '../renderType';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class HurtRockHitAnimation extends State {
  readonly needsBottom = RenderType.POST;

  constructor(context: GameData) { super(context); }

  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndexFast() >= ForestResources.hitRock.length ? HurtRockTalking : null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.hitRock, this.getFrameIndexFast()), 0, 0);
  }
}
